x/valset-pref/validator_set.go

```diff
@@ -117,6 +149,9 @@ func (k Keeper) UndelegateFromValidatorSet(ctx sdk.Context, delegatorAddr string
 			return err
 		}
 
+		// Matt: If we're truncating it here, is it not true that we have the possibility of not undelegating 100%
```

Comment:
I have the same question as delegate here, if we donot truncateInt we cannot how do you recommend getting SharesFromToken which expects an int. 

For all the rounding stuff, i truncated down to make sure we are under delegating/undelegating by 1-2 based denomination than over